src/main/java/net/fabricmc/fabric/api/client/render/EnhancedQuadBakery.java

```diff
@@ -0,0 +1,284 @@
+ */
+
+package net.fabricmc.fabric.api.client.render;
+
+import net.minecraft.client.render.block.BlockRenderLayer;
+import net.minecraft.util.math.Direction;
+
+/**
+ * Helper class to generate vertex data needed to implement EnhancedBakedQuad.<p>
+ * 
+ * For conventional quads, render results will be identical to normal Minecraft. However,
+ * Enhanced Quads support the following additional features:<p>
+ * 
+ * <li>Multiple texture layers. Fabric will render these as separate quads using the
+ * conventional Minecraft render pipeline but advanced rendering mods may render them in a single pass.</li><p>
+ * 
+ * <li>Custom lightmap / emissive rendering. You can specify an additive lightmap for emissive effects.</li><p>
+ * 
+ * <li>Per-layer control of BlockRenderLayer, emissive lightmap, diffuse shading and ambient occlusion.</li><p>
+ * 
+ * <li>Vertex normals.  These are not sent to the BlockRender pipeline (absent a rendering add-on) but are used
+ * to improve diffuse shading for models with non-cube-aligned faces.</li><p>
+ * 
+ * <li>Support for non-square quads.  Uses an improved AO shading algorithm to handle these correctly.</li><p>
+ *
+ * <li>Vertex data includes face and tint usage metadata to simplify EnhancedQuad implementations and
+ * enable performant and consistent application of vertex data.</li><p>
+ * 
+ * Usage:<br>
+ * <li>Obtain an instance. You can and should reuse instances of this class if you are baking multiple quads.</li>
+ * <li>Set vertex and quad attributes, texture depth, render layers, etc.  This can be done in any order.</li>
+ * <li>Call {@link #outputBakedQuadData(int[], int)}</li>
+ * <li>Repeat as needed with changes, optionally calling {@link #clearSettings()} first.</li>  
+ * 
+ */
+public final class EnhancedQuadBakery {
+    /**
+     * Controls alpha and mip-map settings for each texture layer via BlockRenderLayer.<p>
+     * 
+     * Some restrictions apply:<p>
+     * <li>If SOLID occurs it must be first. Only one SOLID layer is possible. (Anything else would be pointless.)</li>
+     * <li>TRANSLUCENT layers must be "above" SOLID and CUTOUT layers.</li>
+     */
+    public final void setRenderLayer(TextureDepth textureLayer, BlockRenderLayer layer) {
+        
+    }
+    
+    /**
+     * Sets texture dept for next output quad. Changing does not clear values 
+     * for unused layers - higher layers not used simply won't be included in output.<p>
+     * 
+     * SINGLE by default. Change applies to the next output quad and remains 
+     * in effect for all subsequent quads until changed.
+     */
+    public final void setTextureDepth(TextureDepth textureDepth) {
+        
+    }
+    
+    public final void position(int vertexIndex, float x, float y, float z) {
+        
+    }
+    
+    public final void normal(int vertexIndex, float x, float y, float z) {
+        
+    }
+    
+    public final void uv(int vertexIndex, TextureDepth textureDepth, float u, float v) {
```

Comment:
Will need to add way to set sprite per layer and document UV coordinates are 0-16.  Should probably also allow rotation and maybe lockUV.  Should basically handle texture interpolation for you similar to what vanilla baking offers.